src/pipert2/core/managers/network.py

```diff
@@ -1,7 +1,7 @@
 from typing import Tuple
 from abc import ABC, abstractmethod
 from collections import defaultdict
-from src.pipert2.core.base.routine import Routine
+from src.pipert2.core.base.routines.routine import Routine
```

Comment:
A comment for all of this kind of imports, if you decided that the abstract class of the routine will be called wrapper routine then change the import to that, instead of importing one of the implementations of the wrapper routine.